src/System.Devices.Gpio/System.Devices.Gpio.Samples/Bme280.cs

```diff
@@ -0,0 +1,419 @@
+                            (_cal_data.dig_H5 * v_x1_u32r) + 16384) >> 15) *
+                         (((((((v_x1_u32r * _cal_data.dig_H6) >> 10) *
+                              (((v_x1_u32r * _cal_data.dig_H3) >> 11) + 32768)) >> 10) +
+                            2097152) * _cal_data.dig_H2 + 8192) >> 14);
+
+            v_x1_u32r = v_x1_u32r - (((((v_x1_u32r >> 15) * (v_x1_u32r >> 15)) >> 7) *
+                                       _cal_data.dig_H1) >> 4);
+
+            v_x1_u32r = (v_x1_u32r < 0) ? 0 : v_x1_u32r;
+            v_x1_u32r = (v_x1_u32r > 419430400) ? 419430400 : v_x1_u32r;
+
+            float h = v_x1_u32r >> 12;
+            _humidity = h / 1024.0f;
+        }
+
+        /// <summary>
+        /// Read the values that are programmed into the sensor during manufacture
+        /// </summary>
+        private void ReadSensorCoefficients()
+        {
+            _cal_data.dig_T1 = Read16LittleEndian(BME280_DIG_T1_REG);
+            _cal_data.dig_T2 = ReadS16LittleEndian(BME280_DIG_T2_REG);
+            _cal_data.dig_T3 = ReadS16LittleEndian(BME280_DIG_T3_REG);
+            _cal_data.dig_P1 = Read16LittleEndian(BME280_DIG_P1_REG);
+            _cal_data.dig_P2 = ReadS16LittleEndian(BME280_DIG_P2_REG);
+            _cal_data.dig_P3 = ReadS16LittleEndian(BME280_DIG_P3_REG);
+            _cal_data.dig_P4 = ReadS16LittleEndian(BME280_DIG_P4_REG);
+            _cal_data.dig_P5 = ReadS16LittleEndian(BME280_DIG_P5_REG);
+            _cal_data.dig_P6 = ReadS16LittleEndian(BME280_DIG_P6_REG);
+            _cal_data.dig_P7 = ReadS16LittleEndian(BME280_DIG_P7_REG);
+            _cal_data.dig_P8 = ReadS16LittleEndian(BME280_DIG_P8_REG);
+            _cal_data.dig_P9 = ReadS16LittleEndian(BME280_DIG_P9_REG);
+            _cal_data.dig_H1 = Read8(BME280_DIG_H1_REG);
+            _cal_data.dig_H2 = ReadS16LittleEndian(BME280_DIG_H2_REG);
+            _cal_data.dig_H3 = Read8(BME280_DIG_H3_REG);
+            _cal_data.dig_H4 = (short)((Read8(BME280_DIG_H4_REG) << 4) | (Read8(BME280_DIG_H4_REG + 1) & 0xF));
+            _cal_data.dig_H5 = (short)((Read8(BME280_DIG_H5_REG + 1) << 4) | (Read8(BME280_DIG_H5_REG) >> 4));
+            _cal_data.dig_H6 = (sbyte)Read8(BME280_DIG_H6_REG);
+        }
+
+        /// <summary>
+        /// Transfers data over the SPI bus
+        /// </summary>
+        private byte SpiTransfer(byte x = 0)
+        {
+            byte result;
+
+            if (_spiSettings != null)
+            {
+                // Hardware SPI
+                var readBuffer = new byte[1];
+
+                if (x == 0)
+                {
+                    _spiDevice.Read(readBuffer);
+                }
+                else
+                {
+                    var writeBuffer = new byte[] { x };
+
+                    _spiDevice.TransferFullDuplex(writeBuffer, readBuffer);
+                }
+
+                result = readBuffer[0];
+            }
+            else
+            {
+                // Software SPI
```

Comment:
Have you tested the software SPI code path?  I wouldn't expect this to work reliably on a non real time OS

Reply:
Yes, and is not working at all. Don't know why, probably because of what you mentioned. Maybe I should remove that posibility from the code.